c/tskit/tables.c

```diff
@@ -8105,6 +8105,285 @@ tsk_table_collection_subset(
     return ret;
 }
 
+static int
+tsk_check_subset_equality(tsk_table_collection_t *self, tsk_table_collection_t *other,
+    tsk_id_t *node_mapping, tsk_size_t other_num_nodes, tsk_size_t num_shared_nodes)
```

Comment:
I don't think we need the other_num_nodes here either - it's just other->nodes.num_rows, right?